### PR TITLE
lp1767068: Fix memory leak when loading cover art

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1767068

### CODE REVIEW

**Comment by uklotzde** [comment]
Crash when upgrading database schema during test setup on macOS, unrelated.

File: src/library/coverartcache.cpp [hunk 1]
```diff
@@ -146,9 +147,13 @@ CoverArtCache::FutureResult CoverArtCache::loadCover(
 
 // watcher
 void CoverArtCache::coverLoaded() {
-    QFutureWatcher<FutureResult>* watcher;
-    watcher = reinterpret_cast<QFutureWatcher<FutureResult>*>(sender());
-    FutureResult res = watcher->result();
+    FutureResult res;
+    {
```
Comment (daschuer) [inline src/library/coverartcache.cpp]: What does the scope here?

**Review by daschuer — COMMENTED** [review]
LGTM, a really tricky case.

File: src/library/coverartcache.cpp (hunk 1 above)
Comment (uklotzde) [reply src/library/coverartcache.cpp]: To ensure that the watcher is not used anywhere else.

**Comment by uklotzde** [comment]
Cover art was already a suspect when @Be-ing  complained about a memory leak during analysis. Finding this one now was just a lucky hit on first sight :)

**Pull request merged**